workspace/filter-workspace-packages/src/index.ts

```diff
@@ -79,7 +79,7 @@ export async function filterPackagesFromDir (
   opts: FilterPackagesOptions & {
     engineStrict?: boolean
     nodeVersion?: string
-    patterns: string[]
+    patterns?: string[]
```

Comment:
@zkochan I was a bit surprised by this change. Should we fix the tests and make this required again?

Changing this to be optional makes it harder to merge in https://github.com/pnpm/pnpm/pull/8214.